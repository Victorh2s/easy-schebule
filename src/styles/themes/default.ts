export const defaultTheme = {
  Light: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#FFFFFF',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#0f0f0f',
    'link-color': '#15A89F',
    'background-form': '#171717',
    'border-input': '#D2D2D4',
  },
  Dark: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#0f0f0f',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#FFFFFF',
    'link-color': '#15A89F',
    'background-form': '#EDEDEF',
    'border-input': '#D2D2D4',
  },
} as const;
