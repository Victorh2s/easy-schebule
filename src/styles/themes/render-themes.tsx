import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './default';
import { BrowserRouter } from 'react-router-dom';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>{' '}
    </BrowserRouter>
  );
};
