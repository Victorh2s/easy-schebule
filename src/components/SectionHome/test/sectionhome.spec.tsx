import { test } from 'vitest';
import { SectionHome } from '../index';
import { renderTheme } from '../../../styles/themes/render-themes';

describe('SectionHome', () => {
  test('SectionHome renders correctly', () => {
    const { getByText, getByAltText } = renderTheme(<SectionHome />);

    expect(
      getByText(
        'Sabemos como pode ser desafiador acompanhar todas as tarefas do dia a dia, tanto no trabalho quanto nas atividades pessoais. É por isso que criamos o EasySchedule: uma ferramenta intuitiva e eficiente para ajudá-lo a manter tudo sob controle.'
      )
    ).toBeInTheDocument();
    expect(getByText('Comece agora')).toBeInTheDocument();
    expect(getByAltText('img')).toBeInTheDocument();
  });

  test('should match to snapshot', () => {
    const { container } = renderTheme(<SectionHome />);

    expect(container).toMatchSnapshot();
  });
});
