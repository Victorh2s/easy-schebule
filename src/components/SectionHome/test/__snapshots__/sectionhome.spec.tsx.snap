// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionHome > should match to snapshot 1`] = `
.c0 {
  padding: 0 7.5rem;
  width: 100%;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  padding: 0 1.5rem;
  height: 80vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 div {
  max-width: 450px;
}

.c1 div a {
  display: block;
  width: 50%;
  margin-top: 1.5rem;
  padding: 0.5rem 1.5rem;
  border-radius: 2rem;
  border: none;
  cursor: pointer;
  text-align: center;
  background: #5300FF;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
  font-weight: bold;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c1 div a:hover {
  background: #15A89F;
  color: #FFFFFF;
}

.c1 img {
  max-width: 450px;
}

.c1 div + img {
  margin-left: 2rem;
}

@media (max-width:700px) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (max-width:1100px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c1 div {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
  }

  .c1 div a {
    display: block;
    width: 50%;
    margin-top: 1.5rem;
    padding: 0.5rem 1.5rem;
    border-radius: 2rem;
    border: none;
    cursor: pointer;
    text-align: center;
    background: #5300FF;
    -webkit-text-decoration: none;
    text-decoration: none;
    color: white;
    font-weight: bold;
  }

  .c1 img {
    display: none;
  }
}

@media (max-width:700px) {
  .c1 {
    padding: 0 0.5rem;
  }

  .c1 div {
    width: 100%;
  }

  .c1 div h1 {
    font-size: 1rem;
  }

  .c1 div a {
    font-size: 1rem;
  }
}

<div>
  <div
    class="c0"
  >
    <section
      class="c1"
    >
      <div>
        <h1>
          Sabemos como pode ser desafiador acompanhar todas as tarefas do dia a dia, tanto no trabalho quanto nas atividades pessoais. É por isso que criamos o EasySchedule: uma ferramenta intuitiva e eficiente para ajudá-lo a manter tudo sob controle.
        </h1>
        <a
          href="/login"
        >
          Comece agora
        </a>
      </div>
      <img
        alt="img"
        aria-label="img"
        src="/src/assets/Complete.svg"
      />
    </section>
  </div>
   
</div>
`;
